src/Analyzers/CSharp/Analyzers/ConvertProgram/ConvertProgramAnalysis_TopLevelStatements.cs

```diff
@@ -0,0 +1,159 @@
+                return false;
+
+            // If the user renamed the 'args' parameter, we can't convert to top level statements.
+            if (methodDeclaration.ParameterList.Parameters.Count == 1 &&
+                methodDeclaration.ParameterList.Parameters[0].Identifier.ValueText != "args")
+            {
+                return false;
+            }
+
+            // Found a suitable candidate.  See if this matches the entrypoint the compiler has actually chosen.
+            var entryPointMethod = semanticModel.Compilation.GetEntryPoint(cancellationToken);
+            if (entryPointMethod == null)
+                return false;
+
+            var thisMethod = semanticModel.GetDeclaredSymbol(methodDeclaration);
+            if (!entryPointMethod.Equals(thisMethod))
+                return false;
+
+            // We found the entrypoint.  However, we can only effectively convert this to top-level-statements
+            // if the existing type is amenable to that.
+            canConvertToTopLevelStatements = TypeCanBeConverted(entryPointMethod.ContainingType, containingTypeDeclaration);
+            return true;
+        }
+
+        private static bool TypeCanBeConverted(INamedTypeSymbol containingType, TypeDeclarationSyntax typeDeclaration)
+        {
+            // Can't convert if our Program type derives or implements anything special.
+            if (containingType.BaseType?.SpecialType != SpecialType.System_Object)
+                return false;
+
+            if (containingType.AllInterfaces.Length > 0)
+                return false;
+
+            // Too complex to convert many parts to top-level statements.  Just bail on this for now.
+            if (containingType.DeclaringSyntaxReferences.Length > 1)
+                return false;
+
+            // Too complex to support converting a nested type.
+            if (containingType.ContainingType != null)
+                return false;
+
+            // If the type wasn't internal it might have been public and something outside this assembly might be using it.
+            if (containingType.DeclaredAccessibility != Accessibility.Internal)
+                return false;
+
+            // type can't be converted with attributes.
+            if (typeDeclaration.AttributeLists.Count > 0)
+                return false;
+
+            // can't convert doc comments to top level statements.
+            if (typeDeclaration.GetLeadingTrivia().Any(t => t.IsDocComment()))
+                return false;
+
+            // All the members of the type need to be private/static.  And we can only have fields or methods. that's to
+            // ensure that no one else was calling into this type, and that we can convert everything in the type to
+            // either locals or local-functions.
+
+            foreach (var member in typeDeclaration.Members)
+            {
+                // method can't be converted with attributes.
```

Comment:
Nit: Local functions can have attributes, so perhaps expand this comment (I assume the reasoning is that attributes likely mean the method is used in reflection?)

Reply:
good point.

Reply:
updated comment to state this.